### PR TITLE
[WIP] python unit tests: add more tests into existing distance test functions.

### PR DESCRIPTION
**Pull request by PetholzA**
this is not the final commit, just checking out how far the coverage increases

### CODE REVIEW

**Comment by fabratu** [comment]
With `[WIP]` you can mark PRs, which are not final. That makes it more obvious to reviewers.

**Comment by fabratu** [comment]
@PetholzA Please rebase on top of master. So far, the PR has increased the coverage from 39% -> 68% for the distance module. If possible, it would be good to increase the coverage even more.

File: networkit/distance.pyx [hunk 1]
```diff
@@ -870,7 +870,7 @@ cdef class Volume:
 cdef extern from "<networkit/distance/JaccardDistance.hpp>":
 
 	cdef cppclass _JaccardDistance "NetworKit::JaccardDistance":
-		_JaccardDistance(const _Graph& G, const vector[count]& triangles) except +
+		_JaccardDistance(const _Graph& G, const vector[count]& _triangles) except +
```
Comment (angriman) [inline networkit/distance.pyx:873]: This change is unrelated with the tests.

File: networkit/distance.pyx [hunk 2]
```diff
@@ -891,7 +891,7 @@ cdef class JaccardDistance:
 
 	cdef _JaccardDistance* _this
 	cdef Graph _G
-	cdef vector[count] triangles
+	cdef vector[count] _triangles
```
Comment (angriman) [inline networkit/distance.pyx:895]: Same here.

File: networkit/test/test_distance_algorithms.py [hunk 3]
```diff
@@ -12,55 +12,55 @@ def setUp(self):
...
 
-	def testDistanceDiameter(self):
+	def testDiameter(self):
 		D = nk.distance.Diameter(self.LL, nk.distance.DiameterAlgo.ESTIMATED_RANGE, error = 0.1)
 		D.run()
 		D = nk.distance.Diameter(self.LL, nk.distance.DiameterAlgo.ESTIMATED_SAMPLES, nSamples = 5)
 		D.run()
 		D = nk.distance.Diameter(self.LL, nk.distance.DiameterAlgo.EXACT)
 		D.run()
+		self.assertIsInstance(D.getDiameter(),tuple)
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:22]: This only test the result of the last algorithm. I'd suggest splitting this test in 3 tests, one per algorithm.

File: networkit/test/test_distance_algorithms.py [hunk 4]
```diff
@@ -12,55 +12,55 @@ def setUp(self):
...
 		E.getValue(self.LL, 0)
 
-
-	def testDistanceEffectiveDiameter(self):
+	def testEffectiveDiameter(self):
 		algo = nk.distance.EffectiveDiameter(self.L)
 		algo.run()
 		algo = nk.distance.EffectiveDiameter(self.LL)
 		algo.run()
+		self.assertIsInstance(algo.getEffectiveDiameter(), float)
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:33]: Same here.

File: networkit/test/test_distance_algorithms.py [hunk 5]
```diff
@@ -12,55 +12,55 @@ def setUp(self):
...
 
-
-	def testDistanceApproxEffectiveDiameter(self):
+	def testApproxEffectiveDiameter(self):
 		algo = nk.distance.EffectiveDiameterApproximation(self.L)
 		algo.run()
 		algo = nk.distance.EffectiveDiameterApproximation(self.LL)
 		algo.run()
+		self.assertIsInstance(algo.getEffectiveDiameter(),float)
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:40]: Same here.

File: networkit/test/test_distance_algorithms.py [hunk 6]
```diff
@@ -12,55 +12,55 @@ def setUp(self):
...
 
-
-	def testDistanceApproxHopPlot(self):
+	def testApproxHopPlot(self):
 		algo = nk.distance.HopPlotApproximation(self.L)
 		algo.run()
 		algo = nk.distance.HopPlotApproximation(self.LL)
 		algo.run()
+		self.assertIsInstance(algo.getHopPlot(), dict)
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:47]: Same here.

File: networkit/test/test_distance_algorithms.py [hunk 7]
```diff
@@ -12,55 +12,55 @@ def setUp(self):
...
 
-
-	def testDistanceNeighborhoodFunction(self):
+	def testNeighborhoodFunction(self):
 		algo = nk.distance.NeighborhoodFunction(self.L)
 		algo.run()
 		algo = nk.distance.NeighborhoodFunction(self.LL)
 		algo.run()
+		self.assertIsInstance(algo.getNeighborhoodFunction(),list)
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:54]: Same here.

File: networkit/test/test_distance_algorithms.py [hunk 8]
```diff
@@ -12,55 +12,55 @@ def setUp(self):
...
 
-
-	def testDistanceApproxNeighborhoodFunction(self):
+	def testApproxNeighborhoodFunction(self):
 		algo = nk.distance.NeighborhoodFunctionApproximation(self.L)
 		algo.run()
 		algo = nk.distance.NeighborhoodFunctionApproximation(self.LL)
 		algo.run()
+		self.assertIsInstance(algo.getNeighborhoodFunction(),list)
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:61]: Same here.

File: networkit/test/test_distance_algorithms.py [hunk 9]
```diff
@@ -175,12 +183,80 @@ def testMultiTargetSTSP(self):
...
+					algo = nk.distance.Dijkstra(g, source, targets, storeNodesSortedByDistance = True)
+				else:
+					algo = nk.distance.BFS(g, source, targets)
 				algo.setSource(source)	
+				algo.setTarget(targets[nTargets-1])
 				algo.run()
-				self.assertLessEqual(len(algo.getPredecessors()), g.numberOfNodes())	
-				self.assertEqual(len(algo.getDistances()), len(targets))
+				self.assertLessEqual(len(algo.getPredecessors(source)),g.numberOfNodes())
+				self.assertLessEqual(len(targets),len(algo.getDistances()))
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:206]: I think this test is incorrect, it will always be true since the number of targets (3) is less than the number of nodes of the graph (that is the length of the `getDistances` list). It can be removed.

File: networkit/test/test_distance_algorithms.py [hunk 10]
```diff
@@ -175,12 +183,80 @@ def testMultiTargetSTSP(self):
...
+				self.assertLessEqual(len(algo.getPaths(targets[nTargets-1])),g.numberOfNodes())
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:207]: I don't follow what this test is checking, could you please clarify that?

File: networkit/test/test_distance_algorithms.py [hunk 11]
```diff
@@ -175,12 +183,80 @@ def testMultiTargetSTSP(self):
...
+				if g.isWeighted():
+					self.assertLessEqual(len(algo.getNodesSortedByDistance()), g.numberOfNodes())
+				self.assertLessEqual(1.0, algo.numberOfPaths(targets[nTargets-1]))
```
Comment (angriman) [inline networkit/test/test_distance_algorithms.py:210]:
```suggestion
				self.assertGreaterEqual(algo.numberOfPaths(targets[nTargets-1]), 1.0)
```

**Review by angriman — CHANGES_REQUESTED** [review]
